wagtail/search/backends/elasticsearch7.py

```diff
@@ -565,6 +565,9 @@ def _process_lookup(self, field, lookup, value):
                 }
             }
 
+    def _process_match_none(self):
+        return {"bool": {"mustNot": {"match_all": {}}}}
```

Comment:
Note that `mustNot` is deprecated in 7.7.0 and 8.0 in favour of `must_not`, ref:
- https://github.com/elastic/elasticsearch/pull/53125
- https://www.elastic.co/guide/en/elasticsearch/reference/7.17/release-notes-7.7.0.html#deprecation-7.7.0

I'm using 8.17.1 and it seems the deprecation is still there, so perhaps they'll remove it in 9.0 instead.

There's a lot of places where we use `mustNot`, and I think there are other camel case fields that should be in snake case, so I'm happy if we handle this in a separate PR.